### ISSUE TITLE
Add support for named fakes

### ISSUE THREAD

**Issue by thomaslevesque** [FakeItEasy/FakeItEasy#1488]
I just found myself in a situation where I have a collection of fakes, and an assertion fails on one of the fakes, but I don't know which. It would be handy to be able to name fakes, e.g. something like this:

```csharp
var foo1 = A.Fake<IFoo>(options => options.Named("foo1"));
var foo2 = A.Fake<IFoo>(options => options.Named("foo2"));
```

Then we could have an assertion failure message like this:

```
Assertion failed for the following call on <foo1>:
  Blah.IFoo.Bar()
Expected to find it once or more but no calls were made to the fake object.
```

The name could also be used for arguments in calls list. (e.g. `Blah.IFoo.Bar(<baz>)`).

For `A.CollectionOfFake<T>`, we might need to introduce an additional overload that takes the index into account:

```csharp
var foos = A.CollectionOfFake<IFoo>(10, (options, index) => options.Named("foo" + index));
```

**Comment by blairconrad** [comment]
I'm super curious about this. For the collection, I can see it happening. I would like to know how it happened to you _without_ using a collection.

**Comment by thomaslevesque** [comment]
> I would like to know how it happened to you _without_ using a collection.

I had a collection; I just didn't create it with `A.CollectionOfFake<T>`.

**Comment by blairconrad** [comment]
> thomaslevesque added this to the vNext milestone an hour ago

Thanks. I'd accidentally added the PR and fixed that, but forgot to readd this!

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 5.1.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/5.1.0).
